src/resmom/start_exec.c

```diff
@@ -1856,6 +1899,195 @@ generate_pbs_nodefile(job *pjob, char *nodefile, int nodefile_sz,
 
 	return (0);
 }
+
+/**
+ * @brief
+ *	Write a piece of data of size 'data_size' into pipe 'upfds'.
+ *
+ * @param[in]	upfds - pipe descriptor upstream.
+ * @param[in]	data - the data to write
+ * @param[in]	data_size - the size of 'data'
+ *
+ * @return int
+ * @retval  0	- for success
+ * @retval  1	- for failure
+ */
+int
+write_pipe_data(upfds, data, data_size)
+int upfds;
```

Comment:
Need to make this ANSi-fied: write_pipe_data(int upfds, void *data, int data_size)